### PR TITLE
Added event documentation for special exams

### PR DESCRIPTION
**Pull request by pdesjardins**
## [DOC-1994](https://openedx.atlassian.net/browse/DOC-1994)

Added event documentation for special exams, which are proctored exams, practice proctored exams, and timed exams. Addresses https://openedx.atlassian.net/browse/DOC-1994.
### Reviewers
- [ ] Subject matter expert: @douglashall 
- [ ] Subject matter expert: @stroilova 
- [ ] Doc team review (sanity check/copy edit/dev edit): @lamagnifica @catong @srpearce 
### Testing
- [ ] Ran ./run_tests.sh without warnings or errors
### HTML Version - links to two sections in the doc draft:
- [x] http://draft-special-exam-events.readthedocs.org/en/latest/internal_data_formats/event_list.html#d-e-f
- [x] http://draft-special-exam-events.readthedocs.org/en/latest/internal_data_formats/tracking_logs.html#special-exam-events
- [x] http://draft-special-exam-events.readthedocs.org/en/latest/internal_data_formats/tracking_logs.html#special-exam-development-events
### Post-review
- [x] Squash commits

### CODE REVIEW

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 1]
```diff
@@ -0,0 +1,734 @@
+.. _special_exam_events:
+
+=================================
+Timed and Proctored Exam Events
+=================================
+
+Special exams are type of course exam that includes timed exams, proctored
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: ...are ^a type...

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 2]
```diff
@@ -0,0 +1,734 @@
...
+exams, and practice proctored exams. These exam types offer more complicated
+functionality than standard exams and they emit additional events. Special exam
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: "standard exams" is actually now a new categorization as well. Maybe instead of introducing that term, use the wording for graded subsections from http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/grading/configure_assignment_type.html
something along the lines of, ... than other assignment types in graded subsections, and ...

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 3]
```diff
@@ -0,0 +1,734 @@
...
+
+``event`` **Member Fields**:
+
+.. list-table::
+   :widths: 15 15 60
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Details
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:69]: alphabetize the member fields, and make the descriptions complete sentences with punctuation

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 4]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``exam_id``
+     - number
+     - The primary identifier of the exam
+   * - ``exam_content_id``
+     - string
+     - The primary identifier of the subsection that contains the exam
+   * - ``exam_name``
+     - string
+     - The title of the exam in the course page, taken from the subsection
+       title
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: change second "title" to "name" to match http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/developing_course/course_subsections.html#change-a-subsection-name

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 5]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``exam_default_time_limit_mins``
+     - number
+     - The standard amount of time, in minutes, that learners have to complete
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: what does standard mean here? the length of time that all learners (who don't have an allowance) get? or the length of time that this particular user has been specifically designated to get?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 6]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``exam_is_proctored``
+     - Boolean
+     - Indicates whether the special exam is a proctored exam
+   * - ``exam_is_practice_exam``
+     - Boolean
+     - Indicates whether the special exam is a practice proctored exam
+   * - ``exam_is_active``
+     - Boolean
+     - Indicates whether the special exam is currently available for learner
+       attempts
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: that seems like an odd thing to capture in a field.  What questions does this field help answer? do we need to segregate out tests of an allowance period, by team members such as beta testers, prior to other learners getting allowances and actually taking the tests for real??

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 7]
```diff
@@ -0,0 +1,734 @@
...
+     - The primary identifier of the subsection that contains the exam
+   * - ``exam_name``
+     - string
+     - The title of the exam in the course page, taken from the subsection
+       title
+   * - ``exam_default_time_limit_mins``
+     - number
+     - The standard amount of time, in minutes, that learners have to complete
+       the exam
+   * - ``exam_is_proctored``
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: why does the event name have "proctored" in it if this might not be a proctored exam? how do the next two fields work when there is also events named edx.special_exam.practice.allowance.created and edx.special_exam.timed.allowance.created ? this needs careful explanation.

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 5 above)
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: I'd maybe stay away from "standard" and describe in terms of how the course team sets this up in Studio. 
http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/course_features/credit_courses/proctored_exams.html?highlight=timed#create-a-proctored-exam
maybe, The length of time allotted to all learners for the exam.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 8]
```diff
@@ -0,0 +1,734 @@
...
+The server emits this event when a learner chooses to take a proctored exam.
+
+**History**: Added 01 Dec 2015.
+
+``event`` **Member Fields**:
+
+.. list-table::
+   :widths: 15 15 60
+   :header-rows: 1
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: please apply comments to first table above throughout where sensible

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 9]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``exam_is_active``
+     - Boolean
+     - Indicates whether the special exam is currently available for learner
+       attempts
+   * - ``attempt_id``
+     - number
+     - The primary identifier of the exam attempt
+   * - ``attempt_user_id``
+     - number
+     - The primary identifier of the learner taking the exam
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: is this the auth_user.id value? 
http://edx.readthedocs.org/en/latest/internal_data_formats/sql_schema.html#id

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 10]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``attempt_started_at``
+     - datetime
+     - The UTC time stamp of the time at which the learner began taking the
+       exam
+   * - ``attempt_completed_at``
+     - datetime
+     - The UTC time stamp of the time at which the learner began taking the
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: began taking --> finished

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 11]
```diff
@@ -0,0 +1,734 @@
...
+       exam
+   * - ``attempt_code``
+     - string
+     - A generated identifier for the exam attempt
+   * - ``attempt_allowed_time_limit_mins``
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: in what cases would this be different from exam_default_time_limit_mins?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 12]
```diff
@@ -0,0 +1,734 @@
...
+     - number
+     - The amount of time, in minutes, that the learner has to complete
+       the exam
+   * - ``attempt_status``
+     - string
+     - The current state of the exam, for example, ``created``
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: what are the possible values here? the whole event captures learners initiating proctored exams, right? what does this field add, possibly through other values here (could there be a notion of "paused" or some such)?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 13]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``attempt_event_elapsed_time_secs``
+     - number
+     - The number of seconds that passed between the time the learner started
+       the exam and the time the event was emitted
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: this one also make me wonder what question someone is trying to answer that they need this value. also Is seconds really the correct unit of measurement, not microseconds? where does the real actual start time come from?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 14]
```diff
@@ -0,0 +1,734 @@
...
+
+.. _special_exam_allowance_created:
+
+``edx.special_exam.proctored.allowance.created``
+************************************************
+
+The server emits this event when a course team grants an exception to the
+standard exam rules for an individual learner.
+
+**History**: Added 01 Dec 2015.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:59]: I'd move this up to line 24 and delete for the individual events.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 15]
```diff
@@ -0,0 +1,734 @@
...
+     - The number of seconds that passed between the time the learner started
+       the exam and the time the event was emitted
+
+``edx.special_exam.proctored.attempt.declined``
+***********************************************
+
+The server emits this event when a learner chooses to take an exam without
+proctoring.
+
+The ``event`` fields for this event are same as the ``event`` fields for the
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: ...are ^the same...

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 13 above)
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: had an interesting chat with Olga. IF the issue is that we capture the time when a leaner agrees to take the exam "with online proctoring", and then some setup happens, and then another time check happens, it would be helpful to understand whether this event is emitted at the time of the first one, when the agreement happens, or the second one, when the exam time allotment commences, would be helpful here and in the event description above...
references to the events that get compared to one another to determine the value would be good

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 16]
```diff
@@ -0,0 +1,734 @@
...
+see :ref:`special_exam_allowance_created`.
+
+**History**: Added 01 Dec 2015.
+
+.. _special_exam_attempt_created:
+
+``edx.special_exam.proctored.attempt.created``
+**********************************************
+
+The server emits this event when a learner chooses to take a proctored exam.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: see notes on line 175. Is this the time when the learner agrees to take the exam with proctoring? or when they have selected "Confirm User Photo" as the last in ~a billion steps to prepare for proctoring?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 17]
```diff
@@ -0,0 +1,734 @@
...
+
+``edx.special_exam.proctored.attempt.declined``
+***********************************************
+
+The server emits this event when a learner chooses to take an exam without
+proctoring.
+
+The ``event`` fields for this event are same as the ``event`` fields for the
+``edx.special_exam.proctored.attempt.created`` event. For more information, see
+:ref:`special_exam_attempt_created`.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:133]: in that case, there may be fields above that could use separate descriptions for people who are getting proctored and people who aren't , if there are any such differences

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 18]
```diff
@@ -0,0 +1,734 @@
...
+
+**History**: Added 01 Dec 2015.
+
+.. _special_exam_attempt_deleted:
+
+``edx.special_exam.proctored.attempt.deleted``
+**********************************************
+
+The server emits this event when a course team removes an exam attempt record
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: I thought it was edX support only who could do that, but might have misunderstood.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 19]
```diff
@@ -0,0 +1,734 @@
...
+for an individual learner.
+
+The ``event`` fields for this event are same as the ``event`` fields for the
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: are ^the same

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 20]
```diff
@@ -0,0 +1,734 @@
...
+
+.. _special_exam_attempt_download_software_clicked:
+
+``edx.special_exam.proctored.attempt.download_software_clicked``
+****************************************************************
+
+The server emits this event when a learner follows the link to download the
+proctoring software for a proctored exam.
+
+The ``event`` fields for this event are same as the ``event`` fields for the
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: pls search for "are same" and replace with "are the same"

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 21]
```diff
@@ -0,0 +1,734 @@
...
+events provide information about special exams included in courses and
+learners' interactions with those special exams.
+
+For more information about timed exams and proctored exams, see
+:ref:`partnercoursestaff:Timed Exams` and
+:ref:`partnercoursestaff:CA_ProctoredExams_Overview`.
+
+This section includes descriptions of the following events. Some events are
+common to more than one type of special exam.
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:47]: thinking outside the box: 
what if we flowcharted out the interesting points in the process when events are emitted, and/or decisions get made by learners? 

```
+-----+
| box | --> etc.
+-----+
```

Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:47]: another outside the box idea: what if we beefed up the description of what proctored, practice, and timed means slightly, and then grouped each of the three identical events together.

`edx.special_exam.practice.attempt.started, edx.special_exam.proctored.attempt.started, edx.special_exam.timed.attempt.started`
`****************************************************`

```
or

edx.special_exam.*.attempt.started
```

`*****************************************`
The server/browser emits these events when a learner does thus and such for a practice, proctored, or timed exam. The name of the event indicates which exam type the learner was engaged in. 

that would mean calling out any distinctions within the descriptions of the member fields, though.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 22]
```diff
@@ -0,0 +1,734 @@
...
+``event`` **Member Fields**:
+
+.. list-table::
+   :widths: 15 15 60
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Details
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:241]: another way to reduce bulk is to say, 
The blahblah event includes the following event member fields.
- (list)

These fields serve the same purpose for this event as for the x:ref:`other_thoroughly_described` event.

The list could include the exam_id, exam_content_id, exam_name, for starters -- they seem to be common to all of these events -- and might not have anything particularly distinctive across the different exam types or the different event-initiating activities.

http://draft-special-exam-events.readthedocs.org/en/latest/internal_data_formats/tracking_logs.html#load-video-edx-video-loaded

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 23]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``review_attempt_code``
+     - string
+     - A generated identifer for the review of the exam attempt
+   * - ``review_raw_data``
+     - string
+     - Information about the review of the exam
+   * - ``review_status``
+     - string
+     - The result of the review. Values include ``Clean``, ``Suspicious``, and
+       ``Rules Violation``.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:303]: you can have a rules violation and not be disqualified?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 24]
```diff
@@ -0,0 +1,734 @@
...
+   * - ``attempt_user_id``
+     - number
+     - The primary identifier of the learner taking the exam
+   * - ``attempt_started_at``
+     - datetime
+     - The UTC time stamp of the time at which the learner began taking the
+       exam
+   * - ``attempt_completed_at``
+     - datetime
+     - The UTC time stamp of the time at which the learner began taking the
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: fix "began"

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 25]
```diff
@@ -0,0 +1,734 @@
...
+       exam
+   * - ``attempt_code``
+     - string
+     - A generated identifier for the exam attempt
+   * - ``attempt_allowed_time_limit_mins``
+     - number
+     - The amount of time, in minutes, that the learner has to complete
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: past tense

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 26]
```diff
@@ -0,0 +1,734 @@
...
+``edx.special_exam.proctored.attempt.created`` event. For more information, see
+:ref:`special_exam_attempt_created`.
+
+**History**: Added 01 Dec 2015.
+
+.. _special_exam_created:
+
+``edx.special_exam.proctored.created``
+**************************************
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: yet another idea for reducing bulk here: move the instructor only event into the section at the end of this file, with xrefs back and forth to this section that would have learner-focused events

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 27]
```diff
@@ -0,0 +1,734 @@
...
+``edx.special_exam.proctored.created`` event. For more information, see
+:ref:`special_exam_created`.
+
+**History**: Added 01 Dec 2015.
+
+.. _special_exam_updated:
+
+``edx.special_exam.proctored.updated``
+**************************************
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: instructor event

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 28]
```diff
@@ -0,0 +1,734 @@
...
+**History**: Added 01 Dec 2015.
+
+``edx.special_exam.practice.allowance.created``
+***********************************************
+
+The server emits this event when a course team grants an exception to the
+standard exam rules for an individual learner.
+
+This event is equivalent to the ``edx.special-
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: I like that!

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 29]
```diff
@@ -0,0 +1,734 @@
...
+The server emits this event when a course team alters a practice proctored exam
+in Studio.
+
+The ``event`` fields for this event are same as the ``event`` fields for the
+``edx.special_exam.proctored.created`` event. For more information, see
+:ref:`special_exam_created`.
+
+**History**: Added 01 Dec 2015.
+
+``edx.special_exam.timed.allowance.created``
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: ok, niggling question: if all proctored exams are timed exams, do both the proctored AND the timed flavored events get emitted for a single student taking a single exam? are practice exams timed too? (and if so, do they get two sets of events?)

**Comment by lamagnifica** [comment]
@pdesjardins I had a lot of questions, and a few suggestions for reducing bulk, but my first pass is complete.

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 14 above)
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:59]: It does seem repetitive to show the same date in all of the event sections. I put them in each one because I was anticipating future development work. If someone adds an event in this section, it might become difficult to understand. This seemed easier to maintain. What do you think about the maintenance aspect?

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 29 above)
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: No, an exam is either proctored, practice, or timed. They all include the timer, but a proctored exam emits only edx.special_exam.proctored.\* events.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 30]
```diff
@@ -0,0 +1,497 @@
...
+  not send exam records to reviewers.
+
+* Timed exams have a time limit. Learners do not install or use monitoring
+  software when they take a timed exam.
+
+The following diagram shows the workflow for a proctored exam. Practice
+proctored exams follow a similar workflow but do not include a review of the
+exam attempt. Timed exams also follow a similar workflow but do not include any
+proctoring software or review of the exam attempt.
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:34]: I'd put only the first sentence before the image, then the two sentences about differences for other exam types.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 31]
```diff
@@ -0,0 +1,497 @@
...
+.. image:: ../../../shared/images/proctored-exam-process.png
+  :width: 500
+  :alt: A diagram showing the workflow for a proctored exam.
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: is the png not part of this PR? I don't see it when I sync to master

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 14 above)
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:59]: well, some of the older events have a long list of dates, 4+, with specific changes. I don't see a problem with stating up front that all of these events were added on a certain date, and then, if a change is made to one of them, have a history note for just the changed event with a description of what was added.

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 32]
```diff
@@ -0,0 +1,497 @@
...
+``edx.special_exam.proctored.allowance.deleted``, ``edx.special_exam.practice.allowance.deleted``, and ``edx.special_exam.timed.allowance.deleted``
+***************************************************************************************************************************************************
+
+The server emits this event when a course team removes an exception to the
+standard exam rules that was previously granted to an individual learner.
+
+The ``event`` fields for this event are the same as the ``event`` fields for
+the ``edx.special_exam.proctored.allowance.created`` event. For more
+information, see :ref:`special_exam_allowance_created`.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: great

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 33]
```diff
@@ -0,0 +1,497 @@
...
+:ref:`special_exam_attempt_created`.
+
+**History**: Added 01 Dec 2015.
+
+.. _special_exam_attempt_deleted:
+
+``edx.special_exam.proctored.attempt.deleted``, ``edx.special_exam.practice.attempt.deleted``, and ``edx.special_exam.timed.attempt.deleted``
+*********************************************************************************************************************************************
+
+The server emits this event when a course team of edX platform administrator
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: of --> or

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 34]
```diff
@@ -0,0 +1,497 @@
...
+
+**History**: Added 01 Dec 2015.
+
+.. _special_exam_created:
+
+``edx.special_exam.proctored.created``, ``edx.special_exam.practice.created``, and ``edx.special_exam.timed.created``
+*********************************************************************************************************************
+
+The server emits this event when a course team creates a proctored exam in
+Studio.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: Apologies if I missed this before. There is a separate section in this rst for Instructor events -- which usually are generated from interactions with the LMS, like all the learner events. This is a new beast, a Studio-generated event, but I think it prob belongs with the Instructor events...

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 35]
```diff
@@ -0,0 +1,497 @@
...
+the ``edx.special_exam.proctored.created`` event. For more information, see
+:ref:`special_exam_created`.
+
+**History**: Added 01 Dec 2015.
+
+.. _special_exam_updated:
+
+``edx.special_exam.proctored.updated``, ``edx.special_exam.practice.updated``, and ``edx.special_exam.timed.updated``
+*********************************************************************************************************************
+
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: Instructor event

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 36]
```diff
@@ -0,0 +1,497 @@
...
+  software when they take a timed exam.
+
+The following diagram shows the workflow for a proctored exam. Practice
+proctored exams follow a similar workflow but do not include a review of the
+exam attempt. Timed exams also follow a similar workflow but do not include any
+proctoring software or review of the exam attempt.
+
+.. image:: ../../../shared/images/proctored-exam-process.png
+  :width: 500
+  :alt: A diagram showing the workflow for a proctored exam.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:37]: that image is terrific! another nit would be to follow the sentence that introduces it, "The following diagram shows the workflow for a proctored exam." with words and words about how events are emitted at each point in the depicted workflow, just to tie the illustration more tightly with what researchers are trying to do.
I hope you can reuse the illustration in the course team guides too!!!!

**Comment by lamagnifica** [comment]
I think this is really good! a couple of suggestions and a typo. :+1:

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 37]
```diff
@@ -0,0 +1,497 @@
+.. _special_exam_events:
+
+=================================
+Timed and Proctored Exam Events
+=================================
+
+Special exams are a type of course exam that includes timed exams, proctored
+exams, and practice proctored exams. These exam types offer more complicated
+functionality than other assignment types in graded subsections and they emit
```
Comment (stroilova) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: Punctuation? "subsection, and"

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 36 above)
Comment (stroilova) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:37]: Great image!!! Thank you for making this so clear. By the way, am I correct in understanding allowances can only be requested at the start of an exam? After requesting an allowance, does the student get taken back to the exam start page?

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 38]
```diff
@@ -0,0 +1,497 @@
...
+     - Details
+   * - ``allowance_key``
+     - string
+     - The type of allowance granted.
+   * - ``allowance_user_id``
+     - number
+     - The identifier of the learner who has been granted the allowance.
+   * - ``allowance_value``
+     - string
+     - The specific details of the allowance granted.
```
Comment (stroilova) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: @pdesjardins  or @douglashall  Is there a list of allowance values, or are these course-staff defined? (Example research question: what kinds of allowances do students request, and how often? Do students requesting allowances have higher grades?)

File: en_us/data/source/internal_data_formats/special_exam_events.rst [hunk 39]
```diff
@@ -0,0 +1,497 @@
...
+     - number
+     - The primary identifier of the exam.
+   * - ``exam_is_active``
+     - Boolean
+     - Indicates whether the special exam is currently available for learner
+       attempts.
+   * - ``exam_is_practice_exam``
+     - Boolean
+     - Indicates whether the special exam is a practice proctored exam.
+   * - ``exam_is_proctored``
```
Comment (stroilova) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: Perhaps this was already asked separately, but is there a reason "practice"/ "proctored" flags are both in the event name and a separate field? @douglashall  on this design question

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 36 above)
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/special_exam_events.rst:37]: Thanks for reviewing the special exam event doc! You are correct, allowances must be requested before a learner starts an exam. Unfortunately, there is no standard way for learners to request an allowance. Each course team must provide its own contact information. So we can't predict how the request will be made, and there is no way to return learners to the exam after they request allowances.

File: en_us/data/source/internal_data_formats/special_exam_events.rst (hunk 38 above)
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/special_exam_events.rst]: Course teams (or PMs) enter written instructions for the reviewers. So I think this string would be something like "this learner is allowed to have another person present in the room."

**Comment by pdesjardins** [comment]
@douglashall and @stroilova, I am planning to publish the documentation for the special exam events. We have an open user story about verifying the design for special exam events (https://openedx.atlassian.net/browse/PHX-250). I think it makes sense to release the information that we have here, because most of it is verified. Do you have any objections to publishing this PR and updating the documentation if we find and fix design problems? Thanks!

**Comment by pdesjardins** [comment]
Attempting to reopen this PR by adding a comment.
